torch/csrc/jit/init.cpp

```diff
@@ -125,12 +126,17 @@ void initJITBindings(PyObject* module) {
           [](std::shared_ptr<Graph>& g) { return PropagateQuantInfo(g); })
       .def(
           "_jit_pass_insert_observers",
-          [](std::shared_ptr<Graph>& g, py::function pyObserverFunction) {
+          [](std::shared_ptr<script::Module>& moduleObj,
+              py::function pyObserverFunction) {
             // Create a new node that would be used in the insert observer pass:
-            // all observer nodes will be cloned from this one.
-            Node* new_node = g->createPythonOp(
-                THPObjectPtr(pyObserverFunction.release().ptr()), "dd", {});
-            InsertObserverNodes(g, new_node);
+           // all observer nodes will be cloned from this one.
+            Graph g;
+            Node* new_node = g.createPythonOp(
+               THPObjectPtr(pyObserverFunction.release().ptr()), "dd", {});
+            const auto& methods = moduleObj->get_methods();
```

Comment:
We should be only inserting observers in the forward pass. Is there a reason to run over all methods of a module here?